oak_io/src/lib.rs

```diff
@@ -83,14 +83,105 @@ impl<L: Decodable, R: Decodable> Decodable for Either<L, R> {
     }
 }
 
+/// A wrapper struct that holds an init message, plus the handle of a channel from which to read
+/// command messages. This is useful for patterns in which an Oak node needs some data at
+/// initialization time, but then handles commands of a different type in an event-loop pattern.
+#[derive(Clone, Debug, PartialEq)]
+pub struct InitWrapper<Init, Command: Decodable> {
+    init: Init,
+    command_receiver: Receiver<Command>,
+}
+
+/// Implementation of [`Encodable`] for [`InitWrapper`] that encodes the handle of the command
+/// receiver channel in the first handle of the resulting [`Message`].
+impl<Init: Encodable, Command: Encodable + Decodable> Encodable for InitWrapper<Init, Command> {
+    fn encode(&self) -> Result<Message, OakError> {
+        let mut init_message = self.init.encode()?;
+        init_message
+            .handles
+            .insert(0, self.command_receiver.handle.handle);
```

Comment:
Is `0` added here because we encode an `Either`?
Please add a comment

Reply:
The 0 is just the position in which to insert (i.e. at the beginning). Added comment to clarify.